Doctrine/Orm/OrderFilter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Doctrine\Orm;
+
+use Doctrine\ORM\QueryBuilder;
+use Dunglas\ApiBundle\Api\ResourceInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ */
+class OrderFilter implements FilterInterface
+{
+    /**
+     * @var array List of properties by witch the collection can or cannot be ordered.
+     */
+    private $sourceList = [];
+
+    /**
+     * @var bool If true $this#sourcelist is used as a whitelist. Otherwise is used as a blacklist.
+     */
+    private $type = true;
+
+    /**
+     * @param array   $properties List of property names on which the filter will be enabled/disabled.
+     * @param boolean $type       If true properties are passed as a whitelist and as a blacklist otherwise.
+     */
+    public function __construct(array $properties = [], $type = true)
+    {
+        $this->sourceList = $properties;
+        $this->type       = $type;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Order collection by properties. The order of the ordered properties is the same as the order specified in the
+     * query.
+     */
+    public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request)
+    {
+        $this->applyFilter($resource, $queryBuilder, $request->query->get('order'));
+    }
+
+    /**
+     * Create query to order the collection by the properties passed.
+     * For each property passed, if the resource does not have such property or if the order value is different from
+     * `asc` or `desc` (case insensitive), the property is ignored.
+     *
+     * @param ResourceInterface $resource
+     * @param QueryBuilder      $queryBuilder
+     * @param array             $values       Array of properties as key and order value as value.
+     *
+     * @return QueryBuilder
+     */
+    protected function applyFilter(ResourceInterface $resource, QueryBuilder $queryBuilder, array $values)
+    {
+        $metadata      = $queryBuilder->getEntityManager()->getClassMetadata($resource->getEntityClass());
+        $fieldNames    = $metadata->getFieldNames();
+        $filterList    = [];
```

Comment:
I suggest injecting the manager instead: https://github.com/dunglas/DunglasApiBundle/blob/master/Doctrine/Orm/Filter.php#L167

Maybe can you add an `abstract` `BaseFilter`class to share logic between `Filter` and `OrderFilter` classes?